Sources/ApolloCodegenLib/Templates/EnumTemplate.swift

```diff
@@ -1,22 +1,35 @@
 import Foundation
+import ApolloUtils
 
 /// Provides the format to convert a [GraphQL Enum](https://spec.graphql.org/draft/#sec-Enums) into
 /// Swift code.
 struct EnumTemplate: TemplateRenderer {
   /// IR representation of source [GraphQL Enum](https://spec.graphql.org/draft/#sec-Enums).
   let graphqlEnum: GraphQLEnumType
+  /// Shared codegen configuration
+  var config: ReferenceWrapped<ApolloCodegenConfiguration>
 
   var target: TemplateTarget { .schemaFile }
 
   var template: TemplateString {
     TemplateString(
     """
     public enum \(graphqlEnum.name.firstUppercased): String, EnumType {
-      \(graphqlEnum.values.map({
-        "case \($0.name)"
+      \(graphqlEnum.values.compactMap({
+        evaluateDeprecation(graphqlEnumValue: $0, config: config)
       }), separator: "\n")
     }
     """
     )
   }
+
+  private func evaluateDeprecation(
+    graphqlEnumValue: GraphQLEnumValue,
+    config: ReferenceWrapped<ApolloCodegenConfiguration>
+  ) -> String? {
+    switch (config.options.deprecatedEnumCases, graphqlEnumValue.deprecationReason) {
+    case (.exclude, .some): return nil
+    default: return "case \(graphqlEnumValue.name)"
```

Comment:
This made me think of something.

For deprecated cases, we probably should have the deprecated annotation to provide a deprecation warning as well. https://stackoverflow.com/questions/71354470/deprecate-enum-value-in-swift

That warning should also apply to fields marked as deprecated in the schema.

Though some people may not want those warning clogging up their project, even when things are deprecated... So  we should add another configuration option for this. Maybe add `public let warningsOnDeprecatedUsage: Composition`. This would add the `@available(*, deprecated, message: "Deprecated: \(deprecationReason)") to enum cases and deprecated fields if set to `.include`.

Can we add this to the list of config options you're working on?

Reply:
This is a pretty nice suggestion, I think that could be really useful. Yes I'll add it to #2101.